backend/copr_backend/storage.py

```diff
@@ -205,15 +216,74 @@ def init_project(self, dirname, chroot):
         # mentioned by its href, not name
         repository = self._get_repository(chroot)
 
-        distribution = self._distribution_name(chroot, dirname)
-        response = self.client.create_distribution(distribution, repository)
+        public_distribution_name = self._distribution_name(chroot, dirname, devel=False)
+        devel_distribution_name = self._distribution_name(chroot, dirname, devel=True)
+
+        # In any case, we need a distribution to be publicly consumed by users
+        response = self.client.create_distribution(public_distribution_name, repository)
         if not response.ok and "This field must be unique" not in response.text:
             self.log.error("Failed to create a Pulp distribution %s because of %s",
-                           distribution, response.text)
+                           public_distribution_name, response.text)
             return False
 
-        response = self.client.create_publication(repository)
-        return response.ok
+        # If a project enabled the manual createrepo mode, we need to create a
+        # devel distribution to be consumed by other builds within the project
+        if self.devel:
+            response = self.client.create_distribution(devel_distribution_name, repository)
+            if not response.ok and "This field must be unique" not in response.text:
+                self.log.error("Failed to create a Pulp distribution %s because of %s",
+                               devel_distribution_name, response.text)
+                return False
```

Comment:
This is still very tricky :sweat_smile:  If create_distribution() fails, we return False, but if it succeeds, we continue to the next section, which is `if not self.devel`.

Perhaps we should just do `if self.devel else:`?